src/stancjs/stancjs.ml

```diff
@@ -72,21 +71,26 @@ let stan2cpp model_name model_string flags =
                Pedantic_analysis.print_warn_uninitialized mir ;
              if is_flag_set "warn-pedantic" then
                Pedantic_analysis.print_warn_pedantic mir ;
-             (cpp, []) )
+             cpp )
+
+let stanc_stdout = ref ""
+let stanc_stderr = ref ""
 
 let wrap_result = function
-  | Result.Ok (s, k) ->
+  | Result.Ok s ->
       Js.Unsafe.obj
         [| ("result", Js.Unsafe.inject (Js.string s))
-         ; ( "warnings"
-           , Js.Unsafe.inject
-               (Js.array (List.to_array (List.map ~f:Js.string k))) ) |]
+         ; ("warnings", Js.Unsafe.inject (Js.string !stanc_stderr)) |]
   | Error e ->
       Js.Unsafe.obj
         [| ("errors", Js.Unsafe.inject (Array.map ~f:Js.string [|e|]))
-         ; ("warnings", Js.Unsafe.inject Js.array_empty) |]
+         ; ("warnings", Js.Unsafe.inject (Js.string !stanc_stderr)) |]
 
 let stan2cpp_wrapped name code (flags : Js.string_array Js.t Js.opt) =
+  stanc_stdout := "" ;
+  stanc_stderr := "" ;
```

Comment:
We have to reset every time because these are global variables. If we don't, we just accumulate warnings from all previous compilations.

Reply:
A better pattern would be to have the components return warnings and errors rather than printing them. Could we do that instead or would that mess up the timing for a release or something?

Reply:
That was my first plan, but I quickly realized that is a pretty big refactor. Particulary not sure on how to return from the parser for example.